### PR TITLE
improve persist car

### PR DESCRIPTION
**Pull request by TheGreenBandit**
thanks Lord Griffin and tupoya for helping implement this

Original post:
broken somehow, needs fix before implementing 

issues:
1,	'byte': ambiguous symbol (compiling source file src\services\vehicle\persist_car_service.cpp)	BigBaseV2	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\util	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\util\world_model.hpp	line 10

2, 	'initializing': cannot convert from 'const Vector3' to 'Hash'	BigBaseV2	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\services\vehicle	C:\Users\TGB\YimMenuAdditions\BigBaseV2\src\services\vehicle\persist_car_service.cpp	line 275

### CODE REVIEW

**Comment by TheGreenBandit** [comment]
once issues are fixed, this will improve persist car in allowing collision, visibility and invincibility to be edited on attachments and base models

**Comment by TheGreenBandit** [comment]
all that needs adding is invincibility, thanks lord griffin for doing this

**Comment by TheGreenBandit** [comment]
Btw with this u would have to resave all ur jsons, loading them from and older version of yim and then saving them on the newer version

File: BigBaseV2/src/core/data/model_attachment.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: why is this even here? just duplicate of `model_hash` ?

File: BigBaseV2/src/core/data/model_attachment.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: why is this even here? just duplicate of model_hash ?

File: BigBaseV2/src/core/data/model_attachment.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: Fix spelling mistakes and make the json name and the name in the struct match

File: BigBaseV2/src/core/data/model_attachment.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/core/data/model_attachment.hpp]: Fix spelling mistakes and make the json name and the name in the struct match

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp:23]: if you add the keys then atleast use them in the actual code

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp]: No, why is that here?

File: BigBaseV2/src/services/vehicle/persist_car_service.hpp
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.hpp]: why a whole namespace for this? remove that and just put that as static member method like the rest

File: README.md
Comment (xiaoxiao921) [inline README.md]: Revert that README change

File: README.md
Comment (xiaoxiao921) [inline README.md]: Same

File: BigBaseV2/src/services/vehicle/persist_car_service.cpp
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.cpp]: Cast to a proper type, `bool`

File: BigBaseV2/src/services/vehicle/persist_car_service.cpp
Comment (xiaoxiao921) [inline BigBaseV2/src/services/vehicle/persist_car_service.cpp]: Cast to a proper type, `bool`

**Review by xiaoxiao921 — DISMISSED** [review]

**Comment by TheGreenBandit** [comment]
Xiao ill give u write access to repository, i cant get on my pc at all today really because of football practice and school

**Comment by TheGreenBandit** [comment]
Tupoya has fixed the issue, it should be ready for merge

**Comment by TheGreenBandit** [comment]
What happend to the discord btw?

**Comment by tupoy-ya** [comment]
> What happend to the discord btw?

https://github.com/YimMenu/YimMenu/discussions/361#discussioncomment-3734636

It's matrix only now.

**Comment by Yimura** [comment]
Check [this discussion comment](https://github.com/YimMenu/YimMenu/discussions/361#discussioncomment-3732080)

**Comment by TheGreenBandit** [comment]
XiaoXiao remove ur requested changes as everything is resolved and ready to merge

**Comment by GeopJr** [comment]
It doesn't matter, someone has to approve this (probably Yimura) before it gets merged.

**Comment by TheGreenBandit** [comment]
Alr, this will be a nice addition to the menu once approved

**Review by Yimura — DISMISSED** [review]

**Comment by GeopJr** [comment]
Hey @TheGreenBandit, does "Lord Griffin" have a GitHub account or an email so we can set them as a co-author (instead of having them as a comment)?

**Comment by TheGreenBandit** [comment]
His github: https://github.com/navmodder

His discord: Peter Griffin#3056

**Review by GeopJr — APPROVED** [review]

**Pull request merged**
